packages/openapi/src/decorators.ts

```diff
@@ -214,3 +218,61 @@ export function resolveInfo(program: Program, entity: Namespace): AdditionalInfo
 function omitUndefined<T extends Record<string, unknown>>(data: T): T {
   return Object.fromEntries(Object.entries(data).filter(([k, v]) => v !== undefined)) as any;
 }
+
+function validateObject(context: DecoratorContext, typespecType: TypeSpecValue, entity: Namespace) {
+  const decorator = entity.decorators.find(
+    (d) => d.decorator === $info && d.node === context.decoratorTarget,
+  );
+  const property = decorator?.definition?.parameters.find((p) => p.name === "additionalInfo")?.type;
```

Comment:
you should be able to get the type with `program.resolveTypeReference` instead of diging into the decorator signature